src/node_file.cc

```diff
@@ -2985,34 +2937,29 @@ void BindingData::LegacyMainResolve(const FunctionCallbackInfo<Value>& args) {
   std::string module_path;
   std::string module_base;
 
-  if (!FileURLToPath(env, package_json_url.value(), module_path)) return;
+  if (!FileURLToPath(env, *package_json_url, &module_path)) return;
 
-  if (args.Length() >= 3 && !args[2]->IsNullOrUndefined() &&
-      args[2]->IsString()) {
+  if (args.Length() >= 3 && args[2]->IsString()) {
     Utf8Value utf8_base_path(env->isolate(), args[2].As<String>());
     auto base_url =
         ada::parse<ada::url_aggregator>(utf8_base_path.ToStringView());
 
     if (!base_url) {
-      env->isolate()->ThrowException(
-          ERR_INVALID_URL(env->isolate(), "Invalid URL"));
-
+      THROW_ERR_INVALID_URL(env->isolate(), "Invalid URL");
       return;
     }
 
-    if (!FileURLToPath(env, base_url.value(), module_base)) return;
-  } else {
-    std::string err_arg_message =
-        "The \"base\" argument must be of type string or an instance of URL.";
-    env->isolate()->ThrowException(
-        ERR_INVALID_ARG_TYPE(env->isolate(), err_arg_message.c_str()));
+    if (!FileURLToPath(env, *base_url, &module_base)) return;
+
+    std::string err_module_message = "Cannot find package '" + module_path +
+                                     "' imported from " + module_base;
+    THROW_ERR_MODULE_NOT_FOUND(env, err_module_message.c_str());
```

Comment:
Since the second argument to `THROW_ERR_*` is a format string, we can do some string formatting here (untested code)
```suggestion
    THROW_ERR_MODULE_NOT_FOUND(env, "Cannot find package '%s' imported from %s", module_path, module_base);
```